### PR TITLE
hypershift: build the control-plane image to catch OCP build regressions

### PR DESCRIPTION
**Pull request by ironcladlou**
This PR adds the Dockerfile.control-plane image to our build pipeline. That
image is what's incorporated into the OCP payload build pipeline, so we need
to at least ensure we don't regress the build itself in our PR workflow.

### CODE REVIEW

File: ci-operator/config/openshift/hypershift/openshift-hypershift-main.yaml [hunk 1]
```diff
@@ -21,6 +21,13 @@ images:
       as:
       - registry.ci.openshift.org/openshift/release:golang-1.17
   to: hypershift
+- dockerfile_path: Dockerfile.control-plane
+  from: base
+  inputs:
+    openshift_release_golang-1.17:
+      as:
+      - registry.ci.openshift.org/openshift/release:golang-1.17
+  to: hypershift-control-plane-operator
```
Comment (alvaroaleman) [inline ci-operator/config/openshift/hypershift/openshift-hypershift-main.yaml:30]: We have to exclude this from promotion I think

Comment (ironcladlou) [reply ci-operator/config/openshift/hypershift/openshift-hypershift-main.yaml:30]: Agree, will fix

**Comment by ironcladlou** [comment]
Depends on https://github.com/openshift/hypershift/pull/833 before we can merge

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**